src/Confluent.Kafka/Producer.cs

```diff
@@ -115,6 +116,22 @@ private void ErrorCallback(IntPtr rk, ErrorCode err, string reason, IntPtr opaqu
             errorHandler?.Invoke(KafkaHandle.CreatePossiblyFatalError(err, reason));
         }
 
+        private Func<PartitionRequest<TKey, TValue>, Partition> partitionerHandler;
+        private Librdkafka.PartitionerDelegate partitionerCallbackDelegate;
+        private int PartitionerCallback(IntPtr rkt,
+            IntPtr keydata,
+            UIntPtr keylen,
+            int partition_cnt,
+            IntPtr rkt_opaque,
+            IntPtr msg_opaque)
+        {
+            // Ensure registered handlers are never called as a side-effect of Dispose/Finalize (prevents deadlocks in common scenarios).
+            if (ownedKafkaHandle.IsClosed) { return 0; }
```

Comment:
This effectively returns partition 0, you will want to use RD_KAFKA_PARTITION_UA (-1)